src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -50,42 +52,44 @@ const DiffExprCompute = (props) => {
     const numMetadata = Object.values(properties).filter((o) => o.type === 'metadataCategorical').length;
 
     if (!numMetadata) {
-      setType(ComparisonType.within);
+      dispatch(setComparisonType(ComparisonType.within));
     }
     setHasMetadata(numMetadata > 0);
 
-    // Update the hierarchies if a previously selected set were to be deleted.
-    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
-      if (!cellSetKey) {
-        return null;
-      }
-
-      const splitKey = cellSetKey.split('/')[1];
+    // If any key in group option is deleted, set option to null
```

Comment:
Also, I just tried doing that locally with these changes to see what is going to happen and I noticed that there is this console error appearing:

<img width="1707" alt="Screenshot 2021-01-21 at 10 12 10" src="https://user-images.githubusercontent.com/65414651/105329166-227f6980-5bd1-11eb-9d82-6b1e61523037.png">

Can you look into it and fix it? Errors and warnings on the console is something that needs looking at - ideally we should not have any errors or warnings.

Reply:
It's fired when a non existent key is used to access an antd Tree component (? - it was fired by rc-tree, a library used by antd's Tree.js) e.g.`property[cellSetName]`. Checking for existence is replaced by using `property.hasOwnProperty(cellSetName)`, which doesn't fire the warning.